extensions/eclipse-che-theia-testing-service-ext/src/browser/languages-test-main.ts

```diff
@@ -0,0 +1,170 @@
+    }
+
+    $registerDocumentLinkProvider(handle: number, pluginInfo: PluginInfo, selector: SerializedDocumentFilter[]): void {
+        this.registerPluginWithFeatureHandle(handle, pluginInfo.id, 'documentLinks');
+        super.$registerDocumentLinkProvider(handle, pluginInfo, selector);
+    }
+
+    $registerCodeLensSupport(handle: number, pluginInfo: PluginInfo, selector: SerializedDocumentFilter[], eventHandle: number): void {
+        this.registerPluginWithFeatureHandle(handle, pluginInfo.id, 'codeLenses');
+        super.$registerCodeLensSupport(handle, pluginInfo, selector, eventHandle);
+    }
+
+    $registerOutlineSupport(handle: number, pluginInfo: PluginInfo, selector: SerializedDocumentFilter[]): void {
+        this.registerPluginWithFeatureHandle(handle, pluginInfo.id, 'symbols');
+        super.$registerOutlineSupport(handle, pluginInfo, selector);
+    }
+
+    $registerDocumentFormattingSupport(handle: number, pluginInfo: PluginInfo, selector: SerializedDocumentFilter[]): void {
+        this.registerPluginWithFeatureHandle(handle, pluginInfo.id, 'documentFormattingEdits');
+        super.$registerDocumentFormattingSupport(handle, pluginInfo, selector);
+    }
+
+    $registerRangeFormattingProvider(handle: number, pluginInfo: PluginInfo, selector: SerializedDocumentFilter[]): void {
+        this.registerPluginWithFeatureHandle(handle, pluginInfo.id, 'documentRangeFormattingEdits');
+        super.$registerRangeFormattingProvider(handle, pluginInfo, selector);
+    }
+
+    $registerOnTypeFormattingProvider(handle: number, pluginInfo: PluginInfo, selector: SerializedDocumentFilter[], autoFormatTriggerCharacters: string[]): void {
+        this.registerPluginWithFeatureHandle(handle, pluginInfo.id, 'onTypeFormattingEdits');
+        super.$registerOnTypeFormattingProvider(handle, pluginInfo, selector, autoFormatTriggerCharacters);
+    }
+
+    $registerFoldingRangeProvider(handle: number, pluginInfo: PluginInfo, selector: SerializedDocumentFilter[]): void {
+        this.registerPluginWithFeatureHandle(handle, pluginInfo.id, 'foldingRange');
+        super.$registerFoldingRangeProvider(handle, pluginInfo, selector);
+    }
+
+    $registerDocumentColorProvider(handle: number, pluginInfo: PluginInfo, selector: SerializedDocumentFilter[]): void {
+        this.registerPluginWithFeatureHandle(handle, pluginInfo.id, 'documentColors');
+        super.$registerDocumentColorProvider(handle, pluginInfo, selector);
+    }
+
+    $registerRenameProvider(handle: number, pluginInfo: PluginInfo, selector: SerializedDocumentFilter[], supportsResolveLocation: boolean): void {
+        this.registerPluginWithFeatureHandle(handle, pluginInfo.id, 'renameEdits');
+        super.$registerRenameProvider(handle, pluginInfo, selector, supportsResolveLocation);
+    }
+
+    private registerPluginWithFeatureHandle(handle: number, extensionId: string, newlyRegisteredAction: string): void {
```

Comment:
I would still move this method to the pluginHandleRegistry: it uses implementation details of the registry unnecessarily.